crates/hotshot/src/types/handle.rs

```diff
@@ -1,12 +1,19 @@
 //! Provides an event-streaming handle for a [`SystemContext`] running in the background
 
-use std::{sync::Arc, time::Duration};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
 
 use async_broadcast::{InactiveReceiver, Receiver, Sender};
 use async_compatibility_layer::art::{async_sleep, async_spawn};
 use async_lock::RwLock;
 #[cfg(async_executor_impl = "async-std")]
 use async_std::task::JoinHandle;
+use event_listener::Event as OtherEvent;
```

Comment:
Can we use a different name for this? Perhaps `EventListenerEvent`, I'd even be okay with the specificity of `ShutdownSignalEvent` since that's all we're using it for.